src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Unix.cs

```diff
@@ -75,21 +75,32 @@ internal bool PreferInlineCompletions
         }
 
         private static unsafe void LoadSocketTypeFromHandle(
-            SafeSocketHandle handle, out AddressFamily addressFamily, out SocketType socketType, out ProtocolType protocolType, out bool blocking, out bool isListening)
+            SafeSocketHandle handle, out AddressFamily addressFamily, out SocketType socketType, out ProtocolType protocolType, out bool blocking, out bool isListening, out bool isSocket)
         {
-            // Validate that the supplied handle is indeed a socket.
-            if (Interop.Sys.FStat(handle, out Interop.Sys.FileStatus stat) == -1 ||
-                (stat.Mode & Interop.Sys.FileTypes.S_IFSOCK) != Interop.Sys.FileTypes.S_IFSOCK)
+            if (Interop.Sys.FStat(handle, out Interop.Sys.FileStatus stat) == -1)
```

Comment:
In what situations will fstat fail?  I'm wondering if rather than throwing we should do:
```C#
isSocket =
    Interop.Sys.FStat(handle, out Interop.Sys.FileStatus stat) != -1 &&
    (stat.Mode & Interop.Sys.FileTypes.S_IFSOCK) == Interop.Sys.FileTypes.S_IFSOCK;
```

Reply:
I think `fstat` failures mean something is wrong with the handle that will bite the user later if we ignore it here.
I don't have a preference between these options.

Reply:
Ok, let's leave it as you have it for now.